ReactNativeClient/lib/models/Tag.js

```diff
@@ -174,23 +342,131 @@ class Tag extends BaseItem {
 		}
 	}
 
+	static tagPath(tags, tagId) {
+		return nestedPath(tags, tagId);
+	}
+
+	static async moveTag(tagId, parentTagId) {
+		if (tagId === parentTagId
+				|| (await Tag.descendantTagIds(tagId)).includes(parentTagId)) {
+			throw new Error(_('Cannot move tag to this location.'));
+		}
+		if (!parentTagId) parentTagId = '';
+
+		let tag = await Tag.load(tagId);
+		if (!tag) return;
+
+		const oldParentTagId = tag.parent_id;
+
+		// Save new parent id
+		tag.parent_id = parentTagId;
+		tag = await Tag.save(tag, { userSideValidation: true });
```

Comment:
Please don't mutate data, and ideally only save the properties that have changed - `const newTag = await Tag.save({ id: tag.id, parent_id: parentTagId }, { userSideValidation: true });`